packages/bsky/src/services/actor/views.ts

```diff
@@ -10,32 +9,22 @@ import { noMatch, notSoftDeletedClause } from '../../db/util'
 import { Actor } from '../../db/tables/actor'
 import { ImageUriBuilder } from '../../image/uri'
 import { LabelService } from '../label'
-import { ListViewBasic } from '../../lexicon/types/app/bsky/graph/defs'
+import { GraphService } from '../graph'
 
 export class ActorViews {
   constructor(private db: Database, private imgUriBuilder: ImageUriBuilder) {}
 
   services = {
-    label: LabelService.creator(),
+    label: LabelService.creator()(this.db),
+    graph: GraphService.creator(this.imgUriBuilder)(this.db),
   }
 
-  profileDetailed(
-    result: ActorResult,
-    viewer: string | null,
-    opts?: { skipLabels?: boolean; includeSoftDeleted?: boolean },
-  ): Promise<ProfileViewDetailed>
-  profileDetailed(
-    result: ActorResult[],
+  async profilesDetailed(
+    results: ActorResult[],
     viewer: string | null,
     opts?: { skipLabels?: boolean; includeSoftDeleted?: boolean },
-  ): Promise<ProfileViewDetailed[]>
-  async profileDetailed(
-    result: ActorResult | ActorResult[],
-    viewer: string | null,
-    opts?: { skipLabels?: boolean; includeSoftDeleted?: boolean },
-  ): Promise<ProfileViewDetailed | ProfileViewDetailed[]> {
-    const results = Array.isArray(result) ? result : [result]
-    if (results.length === 0) return []
+  ): Promise<Record<string, ProfileViewDetailed>> {
```

Comment:
I really don't think we need to do it here/now, but I don't think it would be out of control to eventually start returning `Map`s from these methods rather than `Record`s.  The main thing I worry about with `Record`s is that `record[key]` is always typed to exist, whereas `map.get(key)` is more honest about the `key` possibly not existing in the map.  Maps are also just a little better suited to dynamic keys under the hood.  Of course there's something nice/simple/familiar about using the object notation, that's the main tradeoff.

Reply:
yeah I'd support - we've had some gotchas related to that & it'd be nice to be forced to reckon with it